### PR TITLE
[PORT] Golden Wheelchairs

### PR DESCRIPTION
**Pull request by VaelophisNyx**
Ports ONLY golden wheelchairs from https://github.com/tgstation/tgstation/pull/50606

# Document the changes in your pull request

You can Craft Gilded Wheelchairs with 10 gold, 50u space lube, and a wheelchair
they move just slightly faster for everyone in general, and have immunity to fire and acid (does not give YOU immunity), but are less resistant to other damage.

If you're a Captain with the Paraplegic quirk, you start with a golden wheelchair by default (**UNTESTED**)*

*I could not get my test server under any circumstance down to removing the restrictions on captain, to let me spawn as a captain to test this. Test merge **Required**
# Spriting

ss (2022-09-15 at 07 17 12)
ss (2022-09-15 at 07 17 20)
ss (2022-09-15 at 07 17 28)
ss (2022-09-15 at 07 17 47)

ss (2022-09-16 at 05 31 34)

# Wiki Documentation

Might need to be mentioned anywhere wheelchairs are mentioned?

# Changelog

:cl:  
rscadd: Added Gilded Wheelchairs, crafted with a wheelchair and 10 gold, and 50u space lube
imageadd: Added Golden Wheelchair sprites
/:cl:

### CODE REVIEW

File: code/modules/vehicles/wheelchair.dm [hunk 1]
```diff
@@ -165,3 +190,25 @@
 /obj/item/wheelchair/Destroy()
 	wheelchair = null
 	. = ..()
+
+/obj/item/wheelchair/golden
+	name = "gilded wheelchair"
+	icon_state = "wheelchair_folded_gold"
+	custom_materials = list(/datum/material/gold = 20000)
+	var/obj/vehicle/ridden/wheelchair/golden/wheelchair2
```
Comment (tattax) [inline code/modules/vehicles/wheelchair.dm:198]: what the var doing

File: code/modules/vehicles/wheelchair.dm [hunk 2]
```diff
@@ -165,3 +190,25 @@
...
+	if(has_buckled_mobs())
+		to_chat(usr, span_warning("You need to unbuckle the passenger from [src] first!"))
+		return FALSE
+	usr.visible_message(span_notice("[usr] collapses [src]."), span_notice("You collapse [src]."))
+	var/obj/item/wheelchair/wheelchair_folded = new /obj/item/wheelchair/golden(get_turf(src))
+	forceMove(wheelchair_folded)
+	wheelchair_folded.desc = "A collapsed [name] that can be carried around." 
+	wheelchair_folded.name = name
+	wheelchair_folded.wheelchair = src
+	usr.put_in_hands(wheelchair_folded)
```
Comment (tattax) [inline code/modules/vehicles/wheelchair.dm:214]: change object destruction and wrench act to drop gold and stuff

Comment (tattax) [inline code/modules/vehicles/wheelchair.dm:214]: probably change the initial proc to hold a var the chair changes into

File: code/modules/vehicles/wheelchair.dm (hunk 1 above)
Comment (VaelophisNyx) [reply code/modules/vehicles/wheelchair.dm:198]: the var doin nothing

thanks for pointing it out

File: code/modules/vehicles/wheelchair.dm (hunk 2 above)
Comment (VaelophisNyx) [reply code/modules/vehicles/wheelchair.dm:214]: yeah I probably should, though the idea of permanently erasing gold is amusing

Comment (VaelophisNyx) [reply code/modules/vehicles/wheelchair.dm:214]: god I'd love to do this but I don't even know where to start

File: code/modules/vehicles/wheelchair.dm (hunk 1 above)
Comment (VaelophisNyx) [reply code/modules/vehicles/wheelchair.dm:198]: OH RIGHT

This var for some reason, somehow, controls which wheelchair is spawned when you *UN*fold the damned thing

Comment (tattax) [reply code/modules/vehicles/wheelchair.dm:198]: the original one does, this one does nothing

Comment (tattax) [reply code/modules/vehicles/wheelchair.dm:198]: since that var keeps track of the wheelchair inside the item

Comment (VaelophisNyx) [reply code/modules/vehicles/wheelchair.dm:198]: I mean without that it somehow bricks and spawns a normal wheelchair...with that it works.

**Comment by JamieD1** [comment]
Make it koko exclusive :kek:

Also Captains/Sec Should not be Paraplegic

**Comment by VaelophisNyx** [comment]
I mean they theoretically could be so, is there a reason not to do this?

**Review by cuackles — APPROVED** [review]
do not make it exclusive to koko, love gramps too but more than them should be able to use it

**Pull request closed without merging**